src/vs/base/common/event.ts

```diff
@@ -720,24 +727,17 @@ export class Emitter<T> {
 			// the driver of this
 
 			if (!this._deliveryQueue) {
-				this._deliveryQueue = new LinkedList();
+				this._deliveryQueue = new EventDeliveryQueue();
```

Comment:
As discussed and definitely stretch: maybe create with a non-shared flag or have a `SimpleEventDiliveryQueue` that handles clear better. Maybe we rather hunt down emitter that get disposed while firing or/and with active subscription